src/triage/component/architect/feature_generators.py

```diff
@@ -560,6 +563,26 @@ def index_column_lookup(self, aggregations, imputed=True):
             for aggregation in aggregations
         )
 
+    def _needs_features(self, aggregation):
+        imputed_table = self._clean_table_name(
+            aggregation.get_table_name(imputed=True)
+        )
+
+        if self._table_exists(imputed_table):
+            check_query = f"select 1 from {aggregation.state_table} " +\
+                f"left join {self.features_schema_name}.{imputed_table} using (entity_id, as_of_date) " +\
+                f"where {self.features_schema_name}.{imputed_table}.entity_id is null limit 1"
+            if list(self.db_engine.execute(check_query)):
```

Comment:
Certainly doesn't matter, but I think _this_ is a good use of one of those fun scalar methods we were just talking about 😸 

```suggestion
            if self.db_engine.execute(check_query).scalar():
```

(That is, `scalar()` here will return either `1` or `None`.)